src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,920 @@
+                              SyntaxFactory.ObjectCreationExpression(
+                                  methodSymbol.GenerateReturnTypeSyntax().WithAdditionalAnnotations(Simplifier.Annotation),
+                                  SyntaxFactory.ArgumentList(),
+                                  initializer: null),
+                               variableName: "list",
+                              out documentUpdateInfo);
+                }
+
+                documentUpdateInfo = default;
+                return false;
+            }
+
+            private bool TryConvertIfInInvocation(
+                InvocationExpressionSyntax invocationExpression,
+                Func<ExpressionSyntax, ExpressionSyntax, StatementSyntax> leafExpressionCreationMethod,
+                ExpressionSyntax initializer,
+                string variableName,
+                out DocumentUpdateInfo documentUpdateInfo)
+            {
+                var parentStatement = invocationExpression.GetAncestorOrThis<StatementSyntax>();
+                if (parentStatement != null)
+                {
+                    if (TryConvertIfInInvocationInternal(
+                        invocationExpression,
+                        parentStatement,
+                        initializer,
+                        variableName,
+                        out var variable,
+                        out var nodesBefore,
+                        out var nodesAfter))
+                    {
+                        if (TryConvertQueryExpression(
+                            expression => leafExpressionCreationMethod(variable, expression),
+                            out var statements))
+                        {
+                            var list = new List<CSharpSyntaxNode>();
+                            list.AddRange(nodesBefore);
+                            list.AddRange(statements);
+                            list.AddRange(nodesAfter);
+                            documentUpdateInfo = new DocumentUpdateInfo(parentStatement, list);
+                            return true;
+                        }
+                    }
+                }
+
+                documentUpdateInfo = default;
+                return false;
+            }
+
+            private bool TryConvertIfInInvocationInternal(
+                InvocationExpressionSyntax invocationExpression,
+                StatementSyntax parentStatement,
+                ExpressionSyntax initializer,
+                string variableName,
+                out ExpressionSyntax variable,
+                out CSharpSyntaxNode[] nodesBefore,
+                out CSharpSyntaxNode[] nodesAfter)
+            {
+                var invocationParent = invocationExpression.Parent;
+                switch (invocationParent.Kind())
+                {
+                    case SyntaxKind.EqualsValueClause:
+                        if (invocationParent.Parent is VariableDeclaratorSyntax variableDeclarator)
```

Comment:
>if (invocationParent.Parent is VariableDeclaratorSyntax variableDeclarator) [](start = 24, length = 75)

As discussed offline, consider limiting this to scenario of a local variable declaration statement with single Declarator